source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -279,6 +279,12 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
 
   latchTime(decoder_callbacks_, DNS_START);
   const bool is_proxying = isProxying();
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.dfp_fail_on_empty_host_header")) {
+    if (headers.Host()->value().getStringView().empty()) {
+      onDnsResolutionFail("no_host");
```

Comment:
I'd be inclined to sendLocalReply bfore trying to do a lookup rather than faking resolution failure, WDYT?

Reply:
I changed to fail the request with a bad request.